File: c2/A_Bus_to_Pénjamo.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int t;
    cin >> t;
    while(t--)
    {
        int n, r;
        cin>>n>>r;
        int arr[n];
        long long res = 0, left = 0;
            for(int i=0; i<n; i++){
                cin>>arr[i];
                res += 2*(arr[i]/2);
                r -= arr[i]/2;
                arr[i] = arr[i] % 2;
                if(arr[i]==1) left ++;
            }
            // int lefts = r*2;
            if(r==left || left < r) res += left;
            else{
                left -= r;
                res += (r-left);
            }
            cout << res << endl;
    }
    return 0;
}

File: c2/A_Bus_to_Pénjamo.java

import java.util.Scanner;

public class A_Bus_to_Pénjamo {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-->0){
            int n = sc.nextInt();
            int r = sc.nextInt();
            int[] ar = new int[n];
            long res = 0, left = 0;
            for(int i=0; i<n; i++){
                ar[i] = sc.nextInt();
                res += 2*(ar[i]/2);
                r -= ar[i]/2;
                ar[i] = ar[i] % 2;
                if(ar[i]==1) left ++;
            }
            // int lefts = r*2;
            if(r==left || left < r) res += left;
            else{
                left -= r;
                res += (r-left);
            }
            System.out.println(res);
        }
    }
}